python/tank/descriptor/descriptor_installed_config.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from __future__ import with_statement
+
+import os
+
+from .descriptor_config import ConfigDescriptor
+from . import constants
+from .. import pipelineconfig_utils
+from .. import LogManager
+from ..util import ShotgunPath
+
+from ..errors import TankNotPipelineConfigurationError, TankFileDoesNotExistError, TankInvalidCoreLocationError
+
+log = LogManager.get_logger(__name__)
+
+
+class InstalledConfigDescriptor(ConfigDescriptor):
+    """
+    Descriptor that describes an installed Toolkit Configuration. An installed configuration
+    is what we otherwise refer to as a classic pipeline configuration, which is a pipeline
+    configuration is that installed in a folder on the network, which contains a copy of the
+    environment files, a copy of core and all the bundles required by that pipeline configuration.
+    It supports localized as well as shared core and as such, the interpreter files can be found
+    inside the configuration folder or alongside the shared core.
+    """
+
+    def __init__(self, io_descriptor):
+        super(InstalledConfigDescriptor, self).__init__(io_descriptor)
+        self._io_descriptor.set_missing_manifest_supported(True)
```

Comment:
As we've united both configuration types behind the same interface, not working around a missing manifest means that we would have to start documenting that any manifest based method might blow up if the file is missing for installed config which didn't make this a requirement. It seemed simpler to simply handle that special case here.